contracts/scripts/dry-run.sh

```
@@ -0,0 +1,167 @@
+
+# ── Helpers ──────────────────────────────────────────────────────────────────
+log() { echo "[$(date -u +%H:%M:%S)] $*"; }
+fail() { echo "ERROR: $*" >&2; exit "${1:-1}"; }
+
+require_cmd() {
+  command -v "$1" >/dev/null 2>&1 || fail 3 "Required tool not found: $1"
+}
+
+elapsed_ms() {
+  local start="$1"
+  echo $(( ($(date +%s%3N) - start) ))
```

Shellcheck (0.11.0)

[style] $(( (x) )) is the same as $(( x )). Prefer not wrapping in additional parentheses.

(SC2323)